static/js/settings.js

```diff
@@ -124,6 +124,19 @@ function render_bots() {
     });
 }
 
+exports.generate_zuliprc_uri = function (email, api_key) {
+    var data = settings.generate_zuliprc_content(email, api_key);
+
+    return "data:application/octet-stream;charset=utf-8," + encodeURIComponent(data);
+};
+
+exports.generate_zuliprc_content = function (email, api_key) {
+    return "[api]" +
+           "\nemail=" + email +
+           "\nkey=" + api_key +
+           "\nsite=" + page_params.realm_uri;
```

Comment:
Do we want a trailing newline here?  I think some tools are finicky about not having the last line of content ending with a newline.